src/psyclone/dynamo0p3.py

```diff
@@ -1786,65 +2068,663 @@ def declare_dofmaps(self, parent):
+                dim_name = cma["argument"].name + "_nrow"
+            elif cma["direction"] == "from":
+                dim_name = cma["argument"].name + "_ncol"
+            else:
+                raise InternalError(
+                    "Invalid direction ('{0}') found for CMA operator when "
+                    "collecting indirection dofmaps. Should "
+                    "be either 'to' or 'from'.".format(cma["direction"]))
+            parent.add(DeclGen(parent, datatype="integer", intent="in",
+                               entity_decls=[dim_name]))
+            parent.add(DeclGen(parent, datatype="integer", intent="in",
+                               dimension=dim_name, entity_decls=[dmap]))
 
-    def __init__(self, schedule):
 
-        self._name_space_manager = NameSpaceFactory().create()
+class DynOrientations(DynCollection):
+    '''
+    Handle the declaration of any orientation arrays. Orientation arrays
+    are initialised on a per-cell basis (within the loop over cells) and
+    this is therefore handled by the kernel-call generation.
 
-        # Look at every kernel call in this invoke and generate a set of
-        # the unique CMA operators involved. For each one we create a
-        # dictionary entry. The key is the name of the CMA argument in the
-        # PSy layer and the entry is itself another dictionary containing
-        # two entries: the first 'arg' is the CMA argument object and the
-        # second 'params' is the list of integer variables associated with
-        # that CMA operator. The contents of this list depend on whether
-        # or not the to/from function spaces of the CMA operator are the
-        # same.
-        self._cma_ops = {}
-        for call in schedule.calls():
-            if call.cma_operation:
-                # Get a list of all of the CMA arguments to this call
-                cma_args = psyGen.args_filter(
-                    call.arguments.args,
-                    arg_types=["gh_columnwise_operator"])
-                # Create a dictionary entry for each argument that we
-                # have not already seen
-                for arg in cma_args:
-                    if arg.name not in self._cma_ops:
-                        if arg.function_space_to.orig_name != \
-                           arg.function_space_from.orig_name:
-                            self._cma_ops[arg.name] = {
-                                "arg": arg,
-                                "params": self.cma_diff_fs_params}
-                        else:
-                            self._cma_ops[arg.name] = {
-                                "arg": arg,
-                                "params": self.cma_same_fs_params}
+    '''
+    # We use a named-tuple to manage the storage of the various quantities
+    # that we require. This is neater and more robust than a dict.
+    from collections import namedtuple
+    Orientation = namedtuple("Orientation", ["name", "field",
```

Comment:
Again, why doesn't pycodestyle complain and make use change to "orientation"?

Reply:
Again, Orientation is a type (of namedtuple).